### ISSUE TITLE
Security Issue

### ISSUE THREAD

**Issue by psvitz** [OrchardCMS/OrchardCore#6029]
I may have come across a security issue concerning unauthorized users having access to the system.

Steps:
- log in with a user (establishing a valid session)
- delete this user from system (either from db or using another browser session)
- notice that the deleted user will continue to be able to access the system as long as they have a cookie from a previously authenticated session.

**Comment by deanmarcussen** [comment]
Yes, this is a general ASP.NET Core behaviour / issue with cookie auth.

Related issue https://github.com/OrchardCMS/OrchardCore/issues/1539

However this stackoverflow post has a valid suggestion which we might want to consider implementing as a default / configurable option https://stackoverflow.com/questions/51812100/how-to-remove-cookies-asp-net-core-authorization

(not tested to see if it works, but worth considering)

**Comment by sebastienros** [comment]
In the same way the feature @deanmarcussen mentioned, we could store the timestamp of the last user modification, and invalidate any cookie (when checked) that was issues before that. This should have the same logic as the SecurityStampValidatorOptions options.

**Comment by gvkries** [comment]
This is working as intended. When a user is deleted, their access is revoked the next time the security stamp is validated. Orchard Core uses the default ASP.NET validation interval of 30 minutes, so access is removed after that period.

If needed, this behavior can be changed by adjusting `SecurityStampValidatorOptions.ValidationInterval` in ASP.NET. For more details, see the discussion here:  
<https://github.com/orgs/OrchardCMS/discussions/15717#discussioncomment-9071705>
